pyannote/audio/pipeline/speaker_activity_detection.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+# The MIT License (MIT)
+
+# Copyright (c) 2018-2019 CNRS
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Hervé BREDIN - http://herve.niderb.fr
+
+from typing import Optional
+from pathlib import Path
+import numpy as np
+
+from pyannote.pipeline import Pipeline
+from pyannote.pipeline.parameter import Uniform
+
+from pyannote.core import Annotation, Timeline
+from pyannote.core import SlidingWindowFeature
+
+from pyannote.audio.signal import Binarize
+from pyannote.audio.features import Precomputed
+
+from pyannote.database import get_annotated
+from pyannote.database import get_unique_identifier
+from pyannote.metrics.detection import DetectionPrecision
+from pyannote.metrics.detection import DetectionRecall
+from pyannote.metrics.detection import DetectionErrorRate
+
+
+class SpeakerActivityDetection(Pipeline):
+    """Single speaker activity pipeline
+
+    Parameters
+    ----------
+    scores : `Path`, optional
+        Path to precomputed scores on disk.
+    precision : `float`, optional
+        Target detection precision. Defaults to 0.8.
+    use_der: 'bool', optional
+        Indicates if detection error rate must be use. Default to False
+
+    Hyper-parameters
+    ----------------
+    onset, offset : `float`
+        Onset/offset detection thresholds
+    min_duration_on, min_duration_off : `float`
+        Minimum duration in either state (overlap or not)
+    pad_onset, pad_offset : `float`
+        Padding duration.
+    """
+
+    def __init__(self, label,
```

Comment:
This pipeline is very similar to `pyannote.audio.pipeline.SpeechActivityDetection`.
I'd rather update `SpeechActivityDetection` with support for this kind of task.